app/views/layouts/listnav/_ems_cloud.html.haml

```diff
@@ -40,7 +40,7 @@
             :tables        => 'flavor',
             :controller    => 'ems_cloud')
         - if role_allows(:feature => "security_group_show_list")
-          = li_link(:count => @record.number_of(:security_groups),
+          = li_link(:count => @record.security_groups && @record.number_of(:security_groups),
```

Comment:
Caching like this should be ok. Yeah it handles nil nicely, so no need for that condition.